### ISSUE TITLE
CSS Variable for if the element is overlapping the start or end of the viewport

### ISSUE THREAD

**Issue by notoriousb1t** [scroll-out/scroll-out#8]
@shshaw, continuing our conversation here:
> For scroll out, an intersection variable could work like you described

```css
--intersect-y: -1; // top of viewport
--intersect-y: 0; // fully visible
--intersect-y: 1; // end of viewport
```
>Then you can multiple in your `calc()`   Would also allow for more of a binary approach to specific styles

I think this could work, are you cool with the names --intersect-y / --intersect-x?  That would translate to ```[data-scroll-intersect-x]``` and in the context passed to callbacks, it would be intersectX/Y

**Comment by notoriousb1t** [comment]
Working branch here: https://github.com/scroll-out/scroll-out/tree/intersect-variables

**Comment by notoriousb1t** [comment]
It would be added to onChange/onHidden/onShown and as css variables, but not as a data-attribute.  I have the build wired up here although I haven't had a chance to do anything with it yet:

https://codepen.io/notoriousb1t/pen/gKZGqB/

**Comment by shshaw** [comment]
Demo: https://codepen.io/shshaw/pen/RJEyyR

**Comment by notoriousb1t** [comment]
Published in 2.1.0
